src/js/components/TaskTable.js

```diff
@@ -39,9 +40,55 @@ class TaskTable extends React.Component {
   }
 
   getStatusValue(task) {
+    let taskHealth = this.getTaskHealth(task);
+    if (taskHealth === true) {
+      return 'Healthy';
+    }
+    if (taskHealth === false) {
+      return 'Unhealthy';
+    }
+
     return TaskStates[task.state].displayName;
   }
 
+  getTaskHealth(task) {
+    let mesosTaskHealth = this.getTaskHealthFromMesos(task);
+    if (mesosTaskHealth !== null) {
+      return mesosTaskHealth;
+    }
+
+    return this.getTaskHealthFromMarathon(task);
+  }
+
+  getTaskHealthFromMarathon(task) {
+    const marathonTask = DCOSStore.serviceTree.getTaskFromTaskID(task.id);
+    if (marathonTask != null) {
+      const {healthCheckResults} = marathonTask;
+      if (healthCheckResults != null && healthCheckResults.length > 0) {
+        return healthCheckResults.every(function (result) {
+          return result.alive;
+        });
+      }
+    }
+
+    return null;
+  }
+
+  getTaskHealthFromMesos(task) {
+    if (task.statuses == null) {
+      return null;
+    }
+    const healths = task.statuses.map((status) => status.healthy);
+    const healthDataExists = healths.length > 0 && healths.every(
+      (health) => typeof health !== 'undefined'
+    );
```

Comment:
Thanks for the catch. +1 for killing the rule, I felt mildly resentful as I made this change.